### ISSUE TITLE
Thoughts on getting this into dockerhub as a more generic build?

### ISSUE THREAD

**Issue by chriswhong** [chriswhong/docker-cartodb#23]
Building this image takes over an hour.  I'm contemplating whether it would be good to make it a more generic build which could be `FROM`-able.  Basically, the generic image could be configured to run carto on localhost, without SSL, and would "just work" after pulling the image from Dockerhub and running it.

For advanced configurations, you can make another project that will overwrite all the generic config files and allow you to enable SSL urls, custom domain name, etc.  

Any thoughts on this approach?

**Comment by chriswhong** [comment]
@Prindle19 @pasing @thadk @mgottholsen

**Comment by thadk** [comment]
It does seem to take more time than it did.

This is essentially the only way I got it working the other month -- I had test-built the image for localhost early in the year with a random configuration but never got nginx working. Then in June when I had a use case, but after the build had broken, I realized I could export it and then try/use it on Digital Ocean by going in and altering the baked-in setup.

**Comment by chriswhong** [comment]
Yeah my intent was to keep the manual tinkering to a minimum, the first time I tried this over a year ago I was using a bash terminal in the running container to make changes to config files and it was a mess.

Another thing I am not sure on is whether nginx can run inside the container, and why @sverhoeven didn't do it that way to begin with.  The container can then expose only one port.

**Comment by mgottholsen** [comment]
I'd like to second this. Having a quick "plug and play" version would be very beneficial so people can kick the tires. I think forking it from there with some sort of config script for a more advanced version would be nice for more advanced use cases.

**Comment by brambow** [comment]
Another +1 on this.  

I tried getting this running and kept running into issues related to grunt and the production environment, just like issue #24.  Never quite got everything working.

I ended up installing the image from https://github.com/sverhoeven/docker-cartodb, and it worked. It appears to me that nginx is in the container, as I didn't have to install or configure it separately.  

I like the idea that this image is production-ready, but that seemed to be an obstacle to even getting started with it, at least for me.

**Comment by sindile** [comment]
I came across this repo - https://github.com/codefornl/carto-composer.
Maybe something to consider.

**Comment by rkertesz** [comment]
I saw sverhoeven's post about keeping it simple but I think that it would make sense to set this up the docker way, with a clear separation of concerns for the containers. It looks like the carto-composer repo owner is attempting to do that, no? There is zero documentation for that repo.

**Comment by chriswhong** [comment]
Sorry all, I'm not maintaining this anymore, it served my team well for almost 2 years, but we aren't using it anymore

**Comment by rkertesz** [comment]
@chriswhong Why did you move on to something else and what you are using now which serves your needs better. Is there a blog post or something like that?

**Comment by rkertesz** [comment]
@chriswhong maybe this is the most relevant info as to what you currently use in your stack. would be nice to know more about the mechanics of how the pieces work together. . https://medium.com/nycplanninglabs/our-modern-foss-spatial-stack-9ff2e68a9f8f
